allchblk.c

```diff
@@ -984,6 +989,34 @@ GC_INNER void GC_freehblk(struct hblk *hbp)
           hhdr = prevhdr;
         }
       }
+    #else  /* !defined(__CHERI_PURE_CAPABILITY__) */
+    /* Coalesce with successor, if possible */
+      if(0 != nexthdr && HBLK_IS_FREE(nexthdr) && IS_MAPPED(nexthdr)
```

Comment:
Should that be `nexthdr != NULL` or is it `0` for another reason?

Reply:
The zero comparison in the condition is pre-existing code. 

For the purposes our comparison, I suppose it does not make a difference. 
NULL would be a capability but because the address is 'zero' the  comparison will be satisfied. if a zero is used, the compiler will typecast to 'zero' before doing an integer comparison.

Reply:
Yes, `NULL` and `0` are synonymous on most machines -- but not all! So if we mean `NULL` (which I think we do?) it's best to use `NULL`, for portability reasons if nothing else.

Reply:
In [632aa0c](https://github.com/capablevms/bdwgc/pull/8/commits/632aa0c72a478ccf9fc201842578f7d35b822166). I have changed code to add the capability macros around the *additional capability conditions* only. So this leaves the original code as is. 
So changing  `NULL` to `0`  will change it for other platforms as well. Whichever way you think is better @ltratt

Reply:
I agree that leaving at as `0` makes more sense with 632aa0c.